Update linter to Rome v11.0.0

hotzenklotz opened the pull request:
I was looking for a lint rule to detect unused imports. In the process I update to the latest version of Rome linter v11.0.0 in the hopes that they might have added this rule. Unfortunately, that is not the case, but I updated anyways to have access to their latest rules.
[Rome v11 Announcement and Summary Blog Post](https://rome.tools/blog/2022/12/06/rome11/)

PR changes
- Update Rome to latest version, v11.0.0
- Fix rome config / new sections for some rules
- Disabled new [`noExplicitAny` rule](https://docs.rome.tools/lint/rules/noexplicitany/)
- Applied fixes for new rule [`useFlatMap`](https://docs.rome.tools/lint/rules/useflatmap/) 
- Enabled and applied fixes for rule [`noUnusedVariables`](https://docs.rome.tools/lint/rules/nounusedvariables/ 

### Steps to test:
- yarn install
- yarn lint

### Issues:
- None

------
(Please delete unneeded items, merge only when none are left open)
- [ ] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [(unreleased) migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [ ] Needs datastore update after deployment
- [x] Ready for review

philippotto reviewed (approved):
LGTM :+1: 

> I was looking for a lint rule to detect unused imports. In the process I update to the latest version of Rome linter v11.0.0 in the hopes that they might have added this rule. Unfortunately, that is not the case, but I updated anyways to have access to their latest rules.

Did you try the `noUnusedVariables` rule? I think, it also reports unused imports, but I'm not sure.

hotzenklotz commented:
@philippotto You were right, the `noUnusedVariables` finds unused import. I added it to our config and fixed the 118 errors. Some remarks on the linting errors:
- Ca. 1/3 of errors were unused imports
- Ca 1/3 of errors were `try ... catch (err/exception) ...` that needed a underscore prefix: `catch (_err) ...`
- A few errors where unused properties from object destructuring that I fixed with a `rome-ignore`
- in 1-2 instances the unused variables were in the function signature and unused. in these cases I "refactored" the functions by dropping the unused argument
